code/datums/ai_laws.dm

```diff
@@ -261,6 +266,27 @@
 
 	supplied[number + 1] = law
 
+/datum/ai_laws/proc/replace_random_law(law,groups)
+	var/replaceable_groups = list(LAW_ZEROTH = 0,LAW_ION = 0,LAW_SUPPLIED = 0,LAW_INHERENT = 0)
+	if(zeroth && LAW_ZEROTH in groups)
+		replaceable_groups[LAW_ZEROTH] = 1
+	if(ion.len && LAW_ION in groups)
+		replaceable_groups[LAW_ION] = ion.len
+	if(inherent.len && LAW_INHERENT in groups)
+		replaceable_groups[LAW_INHERENT] = inherent.len
+	if(supplied.len && LAW_SUPPLIED groups)
+		replaceable_groups[LAW_SUPPLIED] = supplied.len

```

Comment:
all of these in statements need () around them. Also the last one is missing the in outright